crates/bevy_winit/src/winit_windows.rs

```diff
@@ -281,52 +283,54 @@ pub fn get_fitting_videomode(
     width: u32,
     height: u32,
 ) -> winit::monitor::VideoMode {
-    let mut modes = monitor.video_modes().collect::<Vec<_>>();
-
     fn abs_diff(a: u32, b: u32) -> u32 {
         if a > b {
             return a - b;
         }
         b - a
     }
 
-    modes.sort_by(|a, b| {
-        use std::cmp::Ordering::*;
-        match abs_diff(a.size().width, width).cmp(&abs_diff(b.size().width, width)) {
-            Equal => {
-                match abs_diff(a.size().height, height).cmp(&abs_diff(b.size().height, height)) {
-                    Equal => b
-                        .refresh_rate_millihertz()
-                        .cmp(&a.refresh_rate_millihertz()),
-                    default => default,
+    monitor
+        .video_modes()
+        .min_by(|a, b| {
+            use std::cmp::Ordering::*;
+            match abs_diff(a.size().width, width).cmp(&abs_diff(b.size().width, width)) {
+                Equal => {
+                    match abs_diff(a.size().height, height).cmp(&abs_diff(b.size().height, height))
+                    {
+                        Equal => b
+                            .refresh_rate_millihertz()
+                            .cmp(&a.refresh_rate_millihertz()),
+                        default => default,
+                    }
                 }
+                default => default,
             }
-            default => default,
-        }
-    });
-
-    modes.first().unwrap().clone()
+        })
+        .expect("no video modes found for monitor")
+        .clone()
 }
 
 /// Gets the "best" videomode from a monitor.
 ///
 /// The heuristic for "best" prioritizes width, height, and refresh rate in that order.
 pub fn get_best_videomode(monitor: &MonitorHandle) -> winit::monitor::VideoMode {
-    let mut modes = monitor.video_modes().collect::<Vec<_>>();
-    modes.sort_by(|a, b| {
-        use std::cmp::Ordering::*;
-        match b.size().width.cmp(&a.size().width) {
-            Equal => match b.size().height.cmp(&a.size().height) {
-                Equal => b
-                    .refresh_rate_millihertz()
-                    .cmp(&a.refresh_rate_millihertz()),
+    monitor
+        .video_modes()
+        .min_by(|a, b| {
+            use std::cmp::Ordering::*;
+            match b.size().width.cmp(&a.size().width) {
+                Equal => match b.size().height.cmp(&a.size().height) {
+                    Equal => b
+                        .refresh_rate_millihertz()
+                        .cmp(&a.refresh_rate_millihertz()),
+                    default => default,
+                },
```

Comment:
Similarly here (except maybe this could use `max_by` and swap the comparison order?)